ios/ErrorConstants.swift

```diff
@@ -0,0 +1,91 @@
+    
+    // MARK: - Error Property Keys
+    static let nameKey = "name"
+    static let messageKey = "message"
+    static let codeKey = "code"
+    static let nativeErrorCodeKey = "nativeErrorCode"
+    
+    // MARK: - Error Name Values
+    static let stripeErrorName = "StripeError"
+    static let nonStripeErrorName = "NonStripeError"
+    
+    // MARK: - ApiError Keys
+    static let apiErrorCodeKey = "code"
+    static let apiErrorMessageKey = "message"
+    static let apiErrorDeclineCodeKey = "declineCode"
+    static let apiErrorChargeKey = "charge"
+    static let apiErrorDocUrlKey = "docUrl"
+    static let apiErrorParamKey = "param"
+    static let apiErrorTypeKey = "type"
+    
+    // MARK: - UnderlyingError Keys
+    static let underlyingErrorCodeKey = "code"
+    static let underlyingErrorMessageKey = "message"
+    static let underlyingErrorIosDomainKey = "iosDomain"
+    static let underlyingErrorIosLocalizedFailureReasonKey = "iosLocalizedFailureReason"
+    static let underlyingErrorIosLocalizedRecoverySuggestionKey = "iosLocalizedRecoverySuggestion"
+    
+    // MARK: - Metadata Keys (iOS-specific from userInfo)
+    // These keys match StripeErrorMetadataKeys in TypeScript
+    static let deviceBannedUntilDateKey = "deviceBannedUntilDate"
+    static let prepareFailedReasonKey = "prepareFailedReason"
+    static let httpStatusCodeKey = "httpStatusCode"
+    static let readerMessageKey = "readerMessage"
+    static let stripeAPIRequestIdKey = "stripeAPIRequestId"
+    static let stripeAPIFailureReasonKey = "stripeAPIFailureReason"
+    static let offlineDeclineReasonKey = "offlineDeclineReason"
+    
+    // MARK: - NSError UserInfo Keys (from Stripe Terminal SDK)
+    // These keys are directly imported from StripeTerminal SDK as ErrorKey enum values
+    // Use .rawValue to access the string key for userInfo dictionary
+    // Reference: https://stripe.dev/stripe-terminal-ios/docs/Errors.html#/c:SCPErrors.h@T@SCPErrorKey
+    
+    // Tap to Pay specific
+    static let deviceBannedUntilDate = ErrorKey.deviceBannedUntilDate.rawValue
+    static let prepareFailedReason = ErrorKey.prepareFailedReason.rawValue
+    
+    // API error related
+    static let httpStatusCode = ErrorKey.httpStatusCode.rawValue
+    static let stripeAPIRequestId = ErrorKey.stripeAPIRequestId.rawValue
+    static let stripeAPIFailureReason = ErrorKey.stripeAPIFailureReason.rawValue
+    static let stripeAPIDeclineCode = ErrorKey.stripeAPIDeclineCode.rawValue
+    static let stripeAPIErrorCode = ErrorKey.stripeAPIErrorCode.rawValue
+    static let stripeAPIErrorType = ErrorKey.stripeAPIErrorType.rawValue
+    static let stripeAPIDocUrl = ErrorKey.stripeAPIDocUrl.rawValue
+    static let stripeAPIErrorParameter = ErrorKey.stripeAPIErrorParameter.rawValue
+    // TODO: Replace with ErrorKey.stripeAPICharge.rawValue when available in Stripe Terminal iOS SDK v5.1+
+    static let stripeAPICharge = "com.stripe-terminal:StripeAPIErrorCharge"
```

Comment:
```suggestion
    static let stripeAPICharge = "com.stripe-terminal:StripeAPICharge"
```

sorry, I was wrong with the first comment...glad I double checked 😅

Reply:
thanks! updated